src/core/namespace.c

```diff
@@ -287,12 +288,19 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
-                break;
+                r = lstat(m->path, &target);
+                if (r != 0)
+                        return -errno;
```

Comment:
The bug is here. `apply_mount` ignores `-`:
```
-bash-4.3# systemd-run --property InaccessiblePaths=-/hey/ho sh -c 'echo HeyHo'
Running as unit: run-rc9e49fca789a4aa98e7f82d08ea33f67.service

-bash-4.3# journalctl -b -u run-rc9e49fca789a4aa98e7f82d08ea33f67.service --no-pager --no-hostname
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Failed to load configuration: No such file or directory
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Trying to enqueue job run-rc9e49fca789a4aa98e7f82d08ea33f67.service/start/fail
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Installed new job run-rc9e49fca789a4aa98e7f82d08ea33f67.service/start as 241
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Enqueued job run-rc9e49fca789a4aa98e7f82d08ea33f67.service/start as 241
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: About to execute: /bin/sh -c 'echo HeyHo'
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Forked /bin/sh as 290
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Changed dead -> running
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Job run-rc9e49fca789a4aa98e7f82d08ea33f67.service/start finished, result=done
Jul 13 09:11:08 systemd[1]: Started /bin/sh -c echo HeyHo.
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Child 290 belongs to run-rc9e49fca789a4aa98e7f82d08ea33f67.service
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Main process exited, code=exited, status=226/NAMESPACE
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Changed running -> failed
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Unit entered failed state.
Jul 13 09:11:08 systemd[1]: run-rc9e49fca789a4aa98e7f82d08ea33f67.service: Failed with result 'exit-code'.


```
This breaks `ProtectHome=yes` if `/run/user` doesn't exists